android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java

```diff
@@ -1,61 +1,79 @@
 package org.fossasia.openevent.fragments;
 
+import android.content.Context;
 import android.content.Intent;
-import android.location.Location;
-import android.location.LocationListener;
 import android.os.Bundle;
-import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
 import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
 import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
 
-import com.google.android.gms.maps.CameraUpdate;
 import com.google.android.gms.maps.CameraUpdateFactory;
 import com.google.android.gms.maps.GoogleMap;
 import com.google.android.gms.maps.OnMapReadyCallback;
 import com.google.android.gms.maps.SupportMapFragment;
 import com.google.android.gms.maps.model.CameraPosition;
+import com.google.android.gms.maps.model.IndoorBuilding;
+import com.google.android.gms.maps.model.IndoorLevel;
 import com.google.android.gms.maps.model.LatLng;
-import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.maps.android.clustering.Cluster;
+import com.google.maps.android.clustering.ClusterManager;
 
 import org.fossasia.openevent.R;
 import org.fossasia.openevent.api.Urls;
-import org.fossasia.openevent.data.Event;
+import org.fossasia.openevent.data.Microlocation;
 import org.fossasia.openevent.dbutils.DbSingleton;
+import org.fossasia.openevent.utils.MyClusterRenderer;
 
-public class MapsFragment extends SupportMapFragment
-        implements LocationListener, OnMapReadyCallback {
+import java.util.ArrayList;
+import java.util.List;
+
+public class MapsFragment extends Fragment
+        implements OnMapReadyCallback {
+
+    private static final double DESTINATION_LATITUDE = 52.52433;
+    private static final double DESTINATION_LONGITUDE = 13.389893;
+    private static final int MAXLEVEL = 3;
```

Comment:
No need of these hardcoded variables. We need  a solution that can go through all the microlocations in the json and create a good looking map.